3d_armor/init.lua

```diff
@@ -436,80 +436,64 @@ end, true)
 
 minetest.register_globalstep(function(dtime)
 	timer = timer + dtime
-	if timer > armor.config.init_delay then
-		for player, count in pairs(pending_players) do
-			local remove = init_player_armor(player) == true
-			pending_players[player] = count + 1
-			if remove == false and count > armor.config.init_times then
-				minetest.log("warning", S("3d_armor: Failed to initialize player"))
-				remove = true
-			end
-			if remove == true then
-				pending_players[player] = nil
+	if timer <= armor.config.init_delay then
+		return
+	end
+	timer = 0
+
+	for player, count in pairs(pending_players) do
+		local remove = init_player_armor(player) == true
+		pending_players[player] = count + 1
+		if remove == false and count > armor.config.init_times then
+			minetest.log("warning", S("3d_armor: Failed to initialize player"))
+			remove = true
+		end
+		if remove == true then
+			pending_players[player] = nil
+		end
+	end
+
+	-- water breathing protection
+	if armor.config.water_protect == true then
+		for _,player in pairs(minetest.get_connected_players()) do
+			local name = player:get_player_name()
```

Comment:
Unnecessary. `get_connected_players` only returns valid players.

Reply:
Isn't this required to scroll through players and reset air/breath if wearing special items ??